### PR TITLE
Code reorg

### PR DESCRIPTION
**Pull request by thakurajayL**

### CODE REVIEW

File: include/common/s1ap_structs.h [hunk 1]
```diff
@@ -39,177 +41,16 @@
 #define NI_DTCH_REQUEST_NO_OF_IES 3
```
Comment (asreek) [inline include/common/s1ap_structs.h:41]: This should be moved as well.

File: include/common/nas_structs.h [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+/* s1apCauseTransport */
+typedef long     s1apCauseTransport_t;
+/* s1apCauseRadioNetwork */
+typedef long     s1apCauseRadioNetwork_t;
+
+
+
+typedef enum emmCause {
+    emmCause_ue_id_not_derived_by_network   = 9,
+    emmCause_network_failure = 11
```
Comment (asreek) [inline include/common/nas_structs.h:148]: network failure cause should be 17.
Need to move nasDetachType enum as well from s1ap_structs.h
(changes in the commit for erab modification)

File: include/common/s1ap_structs.h (hunk 1 above)
Comment (thakurajayL) [reply include/common/s1ap_structs.h:41]: Nope. This is number of IEs in the s1ap protocol. Hence it should remain in s1ap_Strcut file. Let me know @asreek  if you don't agree.

File: include/common/nas_structs.h (hunk 2 above)
Comment (thakurajayL) [reply include/common/nas_structs.h]: Good catch. Fixed this.

**Comment by thakurajayL** [comment]
@yoooou please see why this build fails.

**Comment by thakurajayL** [comment]
retest this please

File: include/common/s1ap_structs.h (hunk 1 above)
Comment (asreek) [reply include/common/s1ap_structs.h:41]: Got it. We can retain it here.

**Comment by yoooou** [comment]
retest this please

**Comment by yoooou** [comment]
> @yoooou please see why this build fails.

It seems TAU test case has been failing consistently for this patch. I see an error from [s1ap log](https://jenkins.opencord.org/job/omec_tc1/919/artifact/mme/s1ap_app.stderr.log). Not sure if it's related

**Comment by asreek** [comment]
> > @yoooou please see why this build fails.
> 
> It seems TAU test case has been failing consistently for this patch. I see an error from [s1ap log](https://jenkins.opencord.org/job/omec_tc1/919/artifact/mme/s1ap_app.stderr.log). Not sure if it's related

@thakurajayL Initial UE TAU got broken when I submitted handover. Fix is in https://github.com/omec-project/Nucleus/pull/37. We may need to merge this first.

**Pull request closed without merging**